src/reduced_order/pod_basis_offline.cpp

```diff
@@ -0,0 +1,156 @@
+#include "pod_basis_offline.h"
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/fe/mapping_q1_eulerian.h>
+
+namespace PHiLiP {
+namespace ProperOrthogonalDecomposition {
+
+template <int dim>
+OfflinePOD<dim>::OfflinePOD(std::shared_ptr<DGBase<dim,double>> &dg_input)
+        : basis(std::make_shared<dealii::TrilinosWrappers::SparseMatrix>())
+        , dg(dg_input)
+        , mpi_communicator(MPI_COMM_WORLD)
+        , mpi_rank(dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD))
+        , pcout(std::cout, mpi_rank==0)
+{
+    const bool compute_dRdW = true;
+    dg_input->assemble_residual(compute_dRdW);
+
+    pcout << "Searching files..." << std::endl;
+
+    getPODBasisFromSnapshots();
+}
+
+template <int dim>
+bool OfflinePOD<dim>::getPODBasisFromSnapshots() {
+    bool file_found = false;
+    MatrixXd snapshotMatrix(0,0);
+    std::string path = dg->all_parameters->reduced_order_param.path_to_search; //Search specified directory for files containing "solutions_table"
+
+    std::vector<std::filesystem::path> files_in_directory;
+    std::copy(std::filesystem::directory_iterator(path), std::filesystem::directory_iterator(), std::back_inserter(files_in_directory));
+    std::sort(files_in_directory.begin(), files_in_directory.end()); //Sort files so that the order is the same as for the sensitivity basis
+
+    for (const auto & entry : files_in_directory){
+        if(std::string(entry.filename()).std::string::find("solution_snapshot") != std::string::npos){
+            pcout << "Processing " << entry << std::endl;
+            file_found = true;
+            std::ifstream myfile(entry);
+            if(!myfile)
+            {
+                pcout << "Error opening file." << std::endl;
```

Comment:
Maybe add std::abort() if file doesn't open (or file_found = false).

Reply:
Fixed in most recent commit.